packages/backend/src/services/DashboardService/DashboardService.ts

```diff
@@ -165,18 +146,38 @@ export class DashboardService {
                 this.spaceModel.getSpaceSummary(spaceUuid),
             ),
         );
+        const dashboardAccesses = await Promise.all(
+            dashboards.map(async (dashboard) => {
+                const spaceAccess = await this.spaceModel.getSpaceAccess(
+                    dashboard.spaceUuid,
+                );
+                return {
+                    uuid: dashboard.uuid,
+                    access: spaceAccess,
+                };
+            }),
+        );
         return dashboards.filter((dashboard) => {
-            const hasAbility = user.ability.can(
-                'view',
-                subject('Dashboard', dashboard),
-            );
             const dashboardSpace = spaces.find(
                 (space) => space.uuid === dashboard.spaceUuid,
             );
+            const spaceAccess = dashboardAccesses.find(
+                (access) => access.uuid === dashboard.uuid,
+            );
+            const hasAbility = user.ability.can(
+                'view',
+                subject('Dashboard', {
+                    ...dashboard,
+                    isPrivate: dashboardSpace?.isPrivate,
+                    access: spaceAccess,
+                }),
+            );
             return (
-                hasAbility &&
                 dashboardSpace &&
-                hasSpaceAccess(user, dashboardSpace, includePrivate)
+                (includePrivate
+                    ? hasAbility
+                    : hasAbility &&
+                      hasDirectAccessToSpace(user, dashboardSpace))
```

Comment:
Took a few minutes to warp my head around this. Seems to be correct but want to manual test.